dll/win32/shell32/shlexec.cpp

```diff
@@ -1863,7 +1863,7 @@ static BOOL SHELL_execute(LPSHELLEXECUTEINFOW sei, SHELL_ExecuteW32 execfunc)
         if(len-1 >= dwApplicationNameLen)
             dwApplicationNameLen = len;
 
-        wszApplicationName = (LPWSTR)HeapAlloc(GetProcessHeap(), 0, dwApplicationNameLen * sizeof(WCHAR));
+        wszApplicationName = (LPWSTR)HeapAlloc(GetProcessHeap(), 0, dwApplicationNameLen * sizeof(WCHAR) + 2);
         memcpy(wszApplicationName, sei_tmp.lpFile + 1, len * sizeof(WCHAR));
 
         if(len > 2)
```

Comment:
Note that here, whatever has been byte-copied above, gets forcibly truncated.

Of course, this code originating from wine, is less than clear about what it does, and does not help code analyzers. It would have been much better to use safe-string copy functions for all those operations.